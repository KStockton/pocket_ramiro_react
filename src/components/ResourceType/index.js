import React, { Component } from 'react';

class ResourceType extends Component {
  constructor() {
    super();
    this.state = {
      category: '',
      company: '',
      contact_number: 0,
      contact_name: ''
    };
  }


  handleChange = (e) => {
    const { name, value } = e.target;
    this.setState({[name]: value});
  }

  handleSubmit = (e) => {
    e.preventDefault()
    const newResource = {
      ...this.state
    };

    this.setState({
      category: '',
      contact_name: '',
      contact_number: 0,
      company: ''
    });
  }
 

  render() {
    const {category, company, contact_name, contact_number} = this.state;


    return (
      <div>
        <section className='nav-wrapper'>
          <input
            id='category'
            type='text' 
            className='part-inputs'
            onChange={this.handleChange}
            value={category}
            name='category'
            placeholder='Enter Category'
          />
          <input
            id='company'
            type='text' 
            className='part-inputs'
            onChange={this.handleChange}
            value={company}
            name='company'
            placeholder='Enter Company'
          />
          <input
            id='contact-name'
            type='text' 
            className='part-inputs'
            onChange={this.handleChange}
            value={contact_name}
            name='contact_name'
            placeholder='Enter contact name'
          />
          <input
            id='contact-number'
            type='number' 
            className='part-inputs'
            onChange={this.handleChange}
            value={contact_number}
            name='contact_number'
            placeholder='Enter Number 3034567890'
          />
          <button onSubmit={this.handleSubmit}>Submit</button>
        </section>
      </div> 
    );
  }
}





export default ResourceType;
