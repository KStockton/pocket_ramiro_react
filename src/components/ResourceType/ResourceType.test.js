import React from 'react';
import { shallow } from 'enzyme';
import ResourceType  from './';  

describe('ResourceType', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ResourceType/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const defaultState = {
      category: '',
      company: '',
      contact_number: 0,
      contact_name: ''
    };
    expect(wrapper.state()).toEqual(defaultState);
  });

  it('should change category in state when handle is invoked', () => {
    wrapper.find('input').at(0).simulate('change', { target: { name: 'category', value: 'cool'}});
    expect(wrapper.state('category')).toBe('cool');
  });

  it('should change company in state when handle is invoked', () => {
    wrapper.find('input').at(1).simulate('change', { target: { name: 'company', value: 'Be Cool'}});
    expect(wrapper.state('company')).toBe('Be Cool');
  });
 
  it('should change contact_name in state when handle is invoked', () => {
    wrapper.find('input').at(2).simulate('change', { target: { name: 'contact_name', value: 'Michael'}});
    expect(wrapper.state('contact_name')).toBe('Michael');
  });

  it('should change contact_number in state when handle is invoked', () => {
    wrapper.find('input').at(3).simulate('change', { target: { name: 'contact_number', value: 3034567654}});
    expect(wrapper.state('contact_number')).toBe(3034567654);
  });

  it('should setState to empty string when handleSubmit is called', () => {
    wrapper.setState({
      category: 'tickets',
      contact_name: 'Mike',
      contact_number: 5469875934,
      company: 'Be Cool' });
    wrapper.update();
    wrapper.instance().handleSubmit({ preventDefault: () => {} });

    const defaultState = {
      category: '',
      company: '',
      contact_number: 0,
      contact_name: ''
    };
    wrapper.update();
    expect(wrapper.state()).toEqual(defaultState);
  });
});