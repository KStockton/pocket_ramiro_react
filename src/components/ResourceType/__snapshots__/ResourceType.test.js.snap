// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceType should match the snapshot 1`] = `
<div>
  <section
    className="nav-wrapper"
  >
    <input
      className="part-inputs"
      id="category"
      name="category"
      onChange={[Function]}
      placeholder="Enter Category"
      type="text"
      value=""
    />
    <input
      className="part-inputs"
      id="company"
      name="company"
      onChange={[Function]}
      placeholder="Enter Company"
      type="text"
      value=""
    />
    <input
      className="part-inputs"
      id="contact-name"
      name="contact_name"
      onChange={[Function]}
      placeholder="Enter contact name"
      type="text"
      value=""
    />
    <input
      className="part-inputs"
      id="contact-number"
      name="contact_number"
      onChange={[Function]}
      placeholder="Enter Number 3034567890"
      type="number"
      value={0}
    />
    <button
      onSubmit={[Function]}
    >
      Submit
    </button>
  </section>
</div>
`;
